### PR TITLE
Feature/fail build on specific metrics

### PR DESCRIPTION
**Pull request by pmckeown**
Added the ability to fail on the number of issues in a given category breaching the defined threshold.

Also documented the release process.

### CODE REVIEW

File: doc/RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Releasing the Dependency Track Maven Plugin
+
+To initiate a new release, raise a Pull Request to the master branch which contains the requested change(s) and sets the 
+POM version number to the next release version.
+
+This project follows semantic versioning principles laid out at [SemVer](https://semver.org/).  As such please follow 
+these guidelines when setting the project version number:
+* Major - Reserved for major releases as defined by the project team or when changes are made to the plugin interface 
+that are not backwards compatible with the previous version.
+* Minor - Backwards compatible changes is to the interface, e.g. a new goal, a new configuration
```
Comment (mterron) [inline doc/RELEASES.md]: extra "is" on '* Minor - Backwards compatible changes is to the interface, e.g. a new goal, a new configuration'

File: src/main/java/io/github/pmckeown/dependencytrack/metrics/MetricsMojo.java [hunk 2]
```diff
@@ -10,11 +10,42 @@
...
+import org.apache.maven.plugins.annotations.Parameter;
 
 import javax.inject.Inject;
 
 /**
- * Provides the capability to print the full set of metrics about a project as determined by the Dependency Track Server
+ * Print the full set of metrics about a project as determined by the Dependency Track Server
+ *
+ * You can optionally define thresholds for failing the build where the number of issues in a particular category
+ * is greater than the threshold you define for that threshold.
```
Comment (mterron) [inline src/main/java/io/github/pmckeown/dependencytrack/metrics/MetricsMojo.java]: "you define for that threshold" -> "you define for that category"

Comment (pmckeown) [reply src/main/java/io/github/pmckeown/dependencytrack/metrics/MetricsMojo.java]: Cheers!

File: doc/RELEASES.md (hunk 1 above)
Comment (pmckeown) [reply doc/RELEASES.md]: Cheers - PR updated.

**Pull request merged**